app/src/main/java/com/khinthirisoe/bakingapp/ui/baking/view/BakingAdapter.kt

```diff
@@ -0,0 +1,76 @@
+package com.khinthirisoe.bakingapp.ui.baking.view
+
+import android.content.Context
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.khinthirisoe.bakingapp.R
+import com.khinthirisoe.bakingapp.data.model.Recipe
+import com.khinthirisoe.bakingapp.ui.base.BaseViewHolder
+import com.khinthirisoe.bakingapp.ui.base.inflate
+import kotlinx.android.synthetic.main.list_baking.view.*
+
+class BakingAdapter(
+    private val context: Context,
+    private var recipes: MutableList<Recipe>?,
+    private val clickListener: BakingRecyclerViewClickListener
+) : RecyclerView.Adapter<BakingAdapter.ViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        return ViewHolder(parent.inflate(R.layout.list_baking))
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.onBind(position)
+    }
+
+    override fun getItemCount(): Int {
+
+        return if (recipes != null && recipes!!.size > 0) {
```

Comment:
@KhinThiriSoe you don't really need to check the size here. What's the point of checking the size if you return zero when the size is not bigger than zero?

Reply:
receipes can be null. if I don't check it, it shows error

Reply:
yes, that's what I mean. you only need to check null